src/cljs/main/broadfcui/page/external_importer.cljs

```diff
@@ -63,46 +161,38 @@
       "Learn how to create a billing project.")]))
 
 (react/defc Importer
-  {:render
+  {:get-initial-state
+   (fn []
+     {:load-status :workspace-check})
+   :render
    (fn [{:keys [props state]}]
      (let [{:keys [source item]} props
            {:keys [load-status error error-response]} @state]
-       [:div {:style style/thin-page-style}
+       [:div {:style {:margin "1.5rem 2rem 0"}}
         [:div {:style {:fontWeight 500 :fontSize "125%" :marginBottom "2rem"}}
          (str "Importing " item " from " source)]
         (case load-status
           :workspace-check (spinner "Checking workspace access...")
-          :version-check (spinner "Loading available method versions...")
+          :error (list
+                  [:div {:style {:marginBottom "1rem" :fontSize "125%"}}
+                   error]
+                  [comps/ErrorViewer {:error error-response}])
           :no-destination (nowhere-to-save-message)
-          :done nil)
-        (when error
-          (list
-           [:div {:style {:marginBottom "1rem" :fontSize "125%"}}
-            error]
-           [comps/ErrorViewer {:error error-response}]))]))
+          :done (case source
+                  "dockstore" (let [[id version] (string/split item #":" 2)]
```

Comment:
this will fail if the version itself contains a colon, right? Probably not worth protecting against right now, we can fix later if we realize it's a problem in practice.

Reply:
Unless dockstore supports versions with colons in them, I guess.